src/VisualStudio/Core/Def/DesignerAttribute/VisualStudioDesignerAttributeService.cs

```diff
@@ -52,11 +52,36 @@ internal class VisualStudioDesignerAttributeService :
         /// </summary>
         private IVSMDDesignerService? _legacyDesignerService;
 
+        /// <summary>
+        /// Queue that tells us to recompute designer attributes when workspace changes happen.  This queue is
+        /// cancellable and will be restarted when new changes come in.  That way we can be quickly update the designer
+        /// attribute for a file when a user edits it.
+        /// </summary>
         private readonly AsyncBatchingWorkQueue _workQueue;
 
-        // We'll get notifications from the OOP server about new attribute arguments. Collect those notifications and
-        // deliver them to VS in batches to prevent flooding the UI thread.
-        private readonly AsyncBatchingWorkQueue<DesignerAttributeData> _projectSystemNotificationQueue;
+        /// <summary>
+        /// We'll get notifications from the OOP server about new attribute arguments. Collect those notifications and
+        /// deliver them to VS in batches to prevent flooding the UI thread.  Importantly, we do not cancel this queue.
+        /// Once we've decided to update the project system, we want to allow that to proceed.
+        /// <para>
+        /// This queue both sends the individual data objects we get back, or the solution instance once we're done with
+        /// a particular project request.  The latter is used so that we can determine which documents are now gone, so
+        /// we can dump our cached data for them.
+        /// </para>
+        /// </summary>
+        private readonly AsyncBatchingWorkQueue<(CodeAnalysis.Solution? solution, DesignerAttributeData? data)> _projectSystemNotificationQueue;
```

Comment:
the PS queue now gets both the data-values, and the last solution we computed against.  with that solution, it can dump old data for documents that are gone.